src/number_theory/bernoulli.lean

```diff
@@ -204,3 +203,36 @@ begin
   { apply factorial_mul_factorial_dvd_factorial_add, },
   { exact cast_ne_zero.mpr hj', },
 end
+
+open ring_hom
+
+/-- Odd Bernoulli numbers (greater than 1) are zero. -/
+theorem bernoulli_odd_eq_zero {n : ℕ} (h_odd : n % 2 = 1) (hlt : 1 < n) : bernoulli n = 0 :=
+begin
+  have f := bernoulli_power_series,
+  have g : eval_neg_hom (mk (λ (n : ℕ), bernoulli n / ↑(n.factorial)) * (exp ℚ - 1)) * (exp ℚ) =
+    (eval_neg_hom (X * exp ℚ)) * (exp ℚ),
+  { congr', },
+  rw [map_mul, map_sub, map_one, map_mul, mul_assoc, sub_mul, mul_assoc (eval_neg_hom X) _ _,
+    mul_comm (eval_neg_hom (exp ℚ)) (exp ℚ), exp_mul_exp_neg_eq_one, eval_neg_hom_X, mul_one,
+    one_mul] at g,
+  suffices h : (mk (λ (n : ℕ), bernoulli n / ↑(n.factorial)) - eval_neg_hom (mk (λ (n : ℕ),
+    bernoulli n / ↑(n.factorial))) ) * (exp ℚ - 1) = X * (exp ℚ - 1),
+  { rw [mul_eq_mul_right_iff] at h, cases h,
+     { simp only [eval_neg_hom, eval_mul_hom, coeff_mk, coe_mk, power_series.ext_iff,
+      coeff_mk, linear_map.map_sub] at h,
+      specialize h n, rw coeff_X n at h, split_ifs at h with h2,
+      { rw h2 at hlt, exfalso, exact false.elim (lt_irrefl _ hlt), },
+      have hn : (n.factorial : ℚ) ≠ 0, { simp [factorial_ne_zero], },
+      rw [←mul_div_assoc, sub_eq_zero_iff_eq, div_eq_iff hn] at h,
+      { rw div_mul_cancel _ hn at h,
+        { simp only [neg_one_pow_eq_pow_mod_two, h_odd, neg_mul_eq_neg_mul_symm, one_mul,
+            pow_one] at h,
+          exact eq_zero_of_neg_eq h.symm, }, }, },
+        { exfalso,
+      rw [power_series.ext_iff] at h,
+      specialize h 1,
+      simpa using h } },
+  { rw [sub_mul, f, mul_sub X, mul_one, sub_right_inj],
```

Comment:
You can still join these two rw like Bryan suggested.

Reply:
In fact, it seems that several other of Bryan's comments you marked as resolved don't appear to be fixed here.

Reply:
Ah, ok, thank you for telling me! I think it is because I pushed something without pulling the previous changes. I will just resolve it.